src/Feeds.php

```diff
@@ -343,8 +343,19 @@ public static function delete_feed( string $feed_id ) {
 			$ad_account_id = Pinterest_For_WooCommerce()::get_setting( 'tracking_advertiser' );
 			APIV5::delete_feed( $feed_id, $ad_account_id );
 			return true;
-		} catch ( Throwable $th ) {
-			Logger::log( $th->getMessage(), 'error' );
+		} catch ( PinterestApiException $e ) {
+			Logger::log( $e->getMessage(), 'error' );
+			if ( in_array(
+				(int) $e->get_pinterest_code(),
+				array(
+					PinterestApiException::MERCHANT_DISAPPROVED,
+					PinterestApiException::MERCHANT_UNDER_REVIEW,
+					PinterestApiException::CATALOGS_FEED_HAS_ACTIVE_PROMOTIONS,
+				)
+			) ) {
+				// Show Admin Notice.
+				FeedDeletionFailure::possibly_add_note( $e->getMessage() );
```

Comment:
We should pass the error code to the note and use it to display predefined message. We don't want to send results from API to the user. We should only log it.